tools/run-dev

```diff
@@ -204,135 +199,79 @@ def start_webpack_watcher() -> "subprocess.Popen[bytes]":
-        pass
-
-    @override
-    def put(self) -> None:
-        pass
-
-    @override
-    def patch(self) -> None:
-        pass
-
-    @override
-    def options(self) -> None:
-        pass
-
-    @override
-    def delete(self) -> None:
-        pass
-
-    @override
-    async def prepare(self) -> None:
-        assert self.request.method is not None
-        assert self.request.remote_ip is not None
-        if "X-REAL-IP" not in self.request.headers:
-            self.request.headers["X-REAL-IP"] = self.request.remote_ip
-        if "X-FORWARDED_PORT" not in self.request.headers:
-            self.request.headers["X-FORWARDED-PORT"] = str(proxy_port)
-        url = transform_url(
-            self.request.protocol,
-            self.request.path,
-            self.request.query,
-            self.target_port,
-            self.target_host,
-        )
-        try:
-            request = httpclient.HTTPRequest(
-                url=url,
-                method=self.request.method,
-                headers=self._add_request_headers(["upgrade-insecure-requests"]),
-                follow_redirects=False,
-                body=self.request.body,
-                allow_nonstandard_methods=True,
-                # use large timeouts to handle polling requests
-                connect_timeout=240.0,
-                request_timeout=240.0,
-                # https://github.com/tornadoweb/tornado/issues/2743
-                decompress_response=False,
-            )
-            response = await client.fetch(request, raise_error=False)
-
-            self.set_status(response.code, response.reason)
-            self._headers = httputil.HTTPHeaders()  # clear tornado default header
-
-            for header, v in response.headers.get_all():
-                # some header appear multiple times, eg 'Set-Cookie'
-                if header.lower() != "transfer-encoding":
-                    self.add_header(header, v)
-            if response.body:
-                self.write(response.body)
-        except (ConnectionError, httpclient.HTTPError) as e:
-            self.set_status(500)
-            self.write("Internal server error:\n" + str(e))
-
-
-class WebPackHandler(BaseHandler):
-    target_port = webpack_port
-
-
-class DjangoHandler(BaseHandler):
-    target_port = django_port
-
-
-class TornadoHandler(BaseHandler):
-    target_port = tornado_port
-
-
-class Application(web.Application):
-    def __init__(self, enable_logging: bool = False) -> None:
-        super().__init__(
-            [
-                (r"/json/events.*", TornadoHandler),
-                (r"/api/v1/events.*", TornadoHandler),
-                (r"/webpack.*", WebPackHandler),
-                (r"/.*", DjangoHandler),
+session: aiohttp.ClientSession
+
+
+HOP_BY_HOP_HEADERS = {
+    hdrs.CONNECTION,
+    hdrs.KEEP_ALIVE,
+    hdrs.PROXY_AUTHENTICATE,
+    hdrs.PROXY_AUTHORIZATION,
+    hdrs.TE,
+    hdrs.TRAILER,
+    hdrs.TRANSFER_ENCODING,
+    hdrs.UPGRADE,
+}
```

Comment:
How did you select this set?

Reply:
Commented (but realistically, if one is going to be messing with this, one had better be taking the time to gain a deeper understanding of what the HTTP specification has to say about it than you’d get from any comment).